### PR TITLE
deprecate Dataproc Operators

### PR DESCRIPTION
**Pull request by Lee-W**

### CODE REVIEW

**Comment by Lee-W** [comment]
image

File: astronomer/providers/google/cloud/operators/dataproc.py [hunk 1]
```diff
@@ -172,61 +51,20 @@ def __init__(
...
-        successful.
-        """
-        if event and event["status"] == "error":
-            raise AirflowException(event["message"])
-        elif event is None:
-            raise AirflowException("No event received in trigger callback")
-        self.log.info("Cluster deleted.")
+        self.polling_interval = polling_interval
+        if self.timeout is None:
+            self.timeout: float = 24 * 60 * 60
```
Comment (pankajkoti) [inline astronomer/providers/google/cloud/operators/dataproc.py:67]: looks like we will need to add a test for this for missing coverage

File: astronomer/providers/google/cloud/operators/dataproc.py [hunk 2]
```diff
@@ -72,98 +25,24 @@ def __init__(
...
-        Relies on trigger to throw an exception, otherwise it assumes execution was
-        successful.
-        """
-        if event and event["status"] == "success":
-            self.log.info("Cluster created successfully \n %s", event["data"])
-            return event["data"]
-        elif event and event["status"] == "error":
-            raise AirflowException(event["message"])
-        raise AirflowException("No event received in trigger callback")
+        kwargs["polling_interval_seconds"] = polling_interval
```
Comment (pankajastro) [inline astronomer/providers/google/cloud/operators/dataproc.py:37]: Nice!

File: astronomer/providers/google/cloud/operators/dataproc.py [hunk 3]
```diff
@@ -172,61 +51,20 @@ def __init__(
...
+            (
+                "This module is deprecated and will be removed in 2.0.0."
+                "Please use `airflow.providers.google.cloud.operators.dataproc.DataprocDeleteClusterOperator`"
+                "and set `deferrable` param to `True` instead."
             ),
-            method_name="execute_complete",
+            DeprecationWarning,
+            stacklevel=2,
         )
+        super().__init__(deferrable=True, **kwargs)
```
Comment (pankajastro) [inline astronomer/providers/google/cloud/operators/dataproc.py:63]: Look like we need to add "polling_interval_seconds" in kwargs here too, no?
https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/operators/dataproc.py#L1002

File: astronomer/providers/google/cloud/operators/dataproc.py [hunk 4]
```diff
@@ -295,61 +100,16 @@ def __init__(
...
+        warnings.warn(
+            (
+                "This module is deprecated and will be removed in 2.0.0."
+                "Please use `airflow.providers.google.cloud.operators.dataproc.DataprocUpdateClusterOperator`"
+                "and set `deferrable` param to `True` instead."
+            ),
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        super().__init__(deferrable=True, **kwargs)
```
Comment (pankajastro) [inline astronomer/providers/google/cloud/operators/dataproc.py:112]: look like we polling_interval_seconds in kwargs https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/operators/dataproc.py#L2452

File: astronomer/providers/google/cloud/operators/dataproc.py (hunk 1 above)
Comment (Lee-W) [reply astronomer/providers/google/cloud/operators/dataproc.py:67]: just resolved. Thanks!

File: astronomer/providers/google/cloud/operators/dataproc.py (hunk 3 above)
Comment (Lee-W) [reply astronomer/providers/google/cloud/operators/dataproc.py:64]: yep, miss this one. just updated it. thanks!

File: astronomer/providers/google/cloud/operators/dataproc.py (hunk 4 above)
Comment (Lee-W) [reply astronomer/providers/google/cloud/operators/dataproc.py:112]: yep, miss this one. just updated it. thanks!

**Comment by Lee-W** [comment]
The mypy here doesn't relate to this PR. I think it fixed by https://github.com/astronomer/astronomer-providers/pull/1455

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**